src/parser.rs

```diff
@@ -1,3 +1,4 @@
+#![allow(deprecated)]
```

Comment:
Unless the warnings are preventing us from doing something with the library, I would keep them on.

Reply:
> Unless the warnings are preventing us from doing something with the library, I would keep them on.

We are using deprecated features of some underlying library, but as this is a known issue _and_ we're going to see things fail if we upgrade these libraries, switching the deprecation messages off is better because it makes feature work and fixes much easier because one does not have to look through pages of useless warnings to find the relevant errors.

Reply:
IMO these warnings should be kept. We are already in the process of migrating off `prec_climber` and the deprecation warnings should not be causing tests and clippy checks to fail.